services/workflows-service/src/collection-flow/collection-flow.service.ts

```diff
@@ -324,4 +327,39 @@ export class CollectionFlowService {
       currentProjectId,
     );
   }
+
+  async uploadNewFile(projectId: string, workflowRuntimeDataId: string, file: Express.Multer.File) {
+    // upload file into a customer folder
+    const customer = await this.customerService.getByProjectId(projectId);
+
+    const runtimeDataId = await this.workflowService.getWorkflowRuntimeDataById(
+      workflowRuntimeDataId,
+      {},
+      [projectId],
+    );
+
+    const entityId = runtimeDataId.businessId || runtimeDataId.endUserId;
+
+    if (!entityId) {
+      throw new NotFoundException("Workflow does't exists");
+    }
+
+    // Remove file extension (get everything before the last dot)
+    const nameWithoutExtension = (file.originalname || generateRandomId(16)).replace(
+      /\.[^.]+$/,
+      '',
+    );
+    // Remove non characters
+    const alphabeticOnlyName = nameWithoutExtension.replace(/\W/g, '');
+
+    const persistedFile = await this.fileService.copyToDestinationAndCreate(
+      { id: alphabeticOnlyName, uri: file.path, provider: 'file' },
```

Comment:
provider is the file provider, in this case of the file is local file system. it should be 'file-system'

Reply:
will change, btw there is no such an implementation for file, this one goes to the default anyway which is local file system